### DISCUSSION TITLE
How to reset EnergyTotal on POW R2 in Tasmota V10?

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#13643] [Q&A] [answered]
Previously I reset the Energy Values of my Sonoff POW R2 using EnergyReset. Since V10, according the Release Notes, this command was removed and replaced by EnergyTotal<x>. Unfortunately this is not working. How shall we handle this command exactly?
I've tried already several things in the console: energytotal 0, energytotal 1, energytotal1 0, energytotal1, ...
All the commands are accepted (no errors or whatsoever) but none of them has reset the desired value. Can someone give some advice here, please?

**Comment by sfromis** [comment, marked as answer]
On the Sonoff Pow R2, there is only one energy monitoring channel/phase, so the index (which defaults to 1) makes no difference. Anyway, did a small test:
```
15:21:04.273 CMD: energytotal
15:21:04.281 MQT: xxxxxxxx/stat/ENERGYTOTAL = {"EnergyTotal":{"Total":135.272,"Yesterday":0.127,"Today":0.115}
15:21:21.573 CMD: energytotal 1
15:21:21.581 MQT: xxxxxxxx/stat/ENERGYTOTAL = {"EnergyTotal":{"Total":0.116,"Yesterday":0.127,"Today":0.115}
```
Appears to work, with the total apparently zeroed to also including the consumption today. Running tasmota.bin.gz v10.0.0. What do you see?

**Reply by ghost** [reply]
My last test (it substracts 1Wh that's all):

15:41:23.135 CMD: energytotal
15:41:23.143 MQT: tasmota/tasmota_F5F861/stat/RESULT = {"EnergyTotal":{"Total":0.013,"Yesterday":0.000,"Today":0.011}
15:41:47.252 CMD: energytotal 1
15:41:47.260 MQT: tasmota/tasmota_F5F861/stat/RESULT = {"EnergyTotal":{"Total":0.012,"Yesterday":0.000,"Today":0.011}
15:41:59.641 CMD: energytotal
15:41:59.649 MQT: tasmota/tasmota_F5F861/stat/RESULT = {"EnergyTotal":{"Total":0.012,"Yesterday":0.000,"Today":0.011}
15:42:13.426 CMD:  energytotal 1
15:42:13.434 MQT: tasmota/tasmota_F5F861/stat/RESULT = {"EnergyTotal":{"Total":0.012,"Yesterday":0.000,"Today":0.011}

**Reply by sfromis** [reply]
Well, as I wrote "zeroed to *also including the consumption today*". If you want to actually test, you could first set it to 9999 and then reset it.

**Reply by ghost** [reply]
OK. Then this is the solution (for me):
energytoday 0
followed by
energytotal 0

Before V10 it was easier: energyreset

Thanks anyway!

**Reply by patman-cp** [reply]
Hah, I had one more I had to zero out, I just guessed the command and ran:

```
energyyesterday 0
```

**Comment by thomas-lentz** [comment]
Hi,
in my opinion the [docs](https://tasmota.github.io/docs/Commands/#power-monitoring) are wrong and incomplete:

**`EnergyTotal<x>`**
Reset Energy Total values
0 = reset
`<value>`= set value
`<time> `= 0..4294967295 set StartTotalTime time as epoch value (optional 2nd parameter)
The new value represents start of day, and output for total includes the today value.

a) there is no blank in the "syntax description" between the `EnergyTotal` and `<x>`
b) `<value>` is not mentioned in the syntax description. Is it meant to be **EnergyTotal `<x>` `<value>`**?
c) where does the time come into play? No syntax no example! Like that **EnergyTotal `<x>` `<value>` `<time>`**?

Thanks to this discussion I was able to reset the counter for total from console.
The value was:
{"Total":-21474.836,"Yesterday":0.000,"Today":0.000}}
after Initial flashing with TASMOTA - the device hasn't been used before. It seems there is a missing initialization (could be also wanted).

Anyway TASMOTA is really **great**! It was only unfortunate this time, that I couldn't use implemented functionalitxy, because of some flaws in the docs.

Greetings
 Tom

**Reply by sfromis** [reply]
There is no blank before `<x>` in the syntax description, because the syntax requires there to be no blank. It does not work correctly if you insert a blank. `<x>` is a very [common syntax element](https://tasmota.github.io/docs/Commands/#how-to-use-commands), an index number placed right after the command name, when the command can address multiple channels. Default is 1, meaning that if you omit it, you get the first channel.

And yes, `<value>` is the number. As in all the other commands, blanks for next parameter. Very plain syntax.

`<time>` is if you want to reset `TotalStartTime`, as visible in the MQTT payloads, to document when the energy monitoring started. It is in the same format as utctime used elsewhere.

Not sure what exactly happens, but in some cases there may be stray data in the part of settings used for energy monitoring, where it indeed is a good plan to start by resetting those.

**Comment by arneman** [comment]
I have to agree with @thomas-lentz that the documentation for EnergyTotal should be amended a little bit. Despite the complete syntax description (`EnergyTotal<x> <value> <time>`) it would also help to describe that <value> has to be Watt - not kwh (even if the result will be displayed in kwh). 

Thanks for your great work! I use Tasmota whenever I can, in at least 15 devices now.

Regards

Arne

**Reply by ascillato** [reply]
Thanks for pointing this out. The documentation is open and maintained by contributions from the community. Please, can you help us and add that to the docs? You can do that by clicking on the pencil at the top right of the page you want to edit. Thanks.

**Reply by mgg1010** [reply]
Hi

I just tested this on 11.1.0.  Typing "EnergyTotal0 0, 1653741642" did not reset the total.   Typing "EnergyTotal 0" did
reset the total.  The docs say it's EnergyTotal<x> with x=0, but for me this doesn't seem to work.

Martin

**Reply by sfromis** [reply]
Nope, that's not what the docs say. The `<x>` part is, like in lots of other commands, an index for which channel it pertains to, like the 3 phases of possible energy monitoring. The index defaults to 1 if not specified. And there is no channel 0 for energy monitoring, hence this cannot work. In a few commands, index 0 means all channels (or whatever), but that's not the case here. One correct form would be `EnergyTotal1` with the relevant value+timestamp or the equivalent `EnergyTotal`.

The `0 = reset` in the command description is about the value, not the index. I've tweaked the docs a bit to make it more clear.

**Reply by mgg1010** [reply]
Docs look clearer now, thansk :)

**Reply by arneman** [reply]
I've just created a pull request, add the unit (Wh) and describe the commands a little but more. I could imagine that it is hard to describe things in a way a Tasmota noob will understand when you already made hundreds of commits and know tasmota for years.

In general I would suggest to add a new column "Description" to the command documentation (or change the design in another way to have a separate section for this), At the moment this is done in the "Parameters" column, what leads to confusion in some cases and makes it in general hard to write and read/understand it.
I'm not feeling comfortable to do these changes because I don't know the project and commands so well like the core team, I don't have the design skills (e.g. has to work on mobiles too) and of course, the time... But maybe you can add this to a wishlist, if you have such? I think the documentation is very important for (new) users and thus for success of Tasmota.

Regards

Arne

**Reply by sfromis** [reply]
My role around the docs is not one of restructuring, I merely do small specific tweaks of simple details, and stay out of larger scale changes.

**Comment by laurentppol** [comment]
Oooups, works on 11.0.0 (POW R2) but ONLY for energy total:
`"energytotal 703819 2021-06-01T00:00:00"`

And on console there is still:

```
19:20:21.189 MQT: tele/tasmota_B8165A/STATE = {"Time":"2022-05-31T19:20:21","Uptime":"0T00:28:44","UptimeSec":1724,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"IoT","BSSId":"24:05:0F:31:3C:D6","Channel":6,"Mode":"11n","RSSI":64,"Signal":-68,"LinkCount":1,"Downtime":"0T00:00:05"}}
19:20:21.201 MQT: tele/tasmota_B8165A/SENSOR = {"Time":"2022-05-31T19:20:21","ENERGY":{"TotalStartTime":"1970-01-01T00:33:41","Total":704.054,"Yesterday":0.000,"Today":0.235,"Period": 4,"Power":219,"ApparentPower":264,"ReactivePower":147,"Factor":0.83,"Voltage":230,"Current":1.150}}
```

So date is at beginning of Unix time :)
After factory-reset caused by power problems.

Could someone confirm / negate result?

**Reply by sfromis** [reply]
The time is in epoch format, not in text format, Unix timestamp value divided by 1000, seconds since 1970. Docs:
> `<time> = 0..4294967295` set StartTotalTime time as epoch value

**Reply by laurentppol** [reply]
THX, omitted that (even it is in current thread).

Now it worked. 
[just my 2 cents] 
Shouldn't there be at least error message? "invalid date format"? (this should be easy to verify in Tasmota code).
No error AND no effect is misleading a bit.

**Reply by sfromis** [reply]
While I see how that would be nice, it would actually be a huge change and memory cost, as input validation would logically also have to apply to all other commands and rules. Currently is is just normal that bad input does not lead to an error message, but that it simply works badly, or does nothing.

**Comment by FARKIr** [comment]
SOME PROBLEM HAVE I ...CAN SOMBODY HELP ME PLEASE ? 

18:23:55.731 CMD: EnergyTotal1 0
18:23:55.741 MQT: stat/tasmota_BC6C11/RESULT = {"EnergyTotal":{"Total":[0.000,3.129,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001]}}
18:23:58.444 CMD: EnergyTotal2 0
18:23:58.453 MQT: stat/tasmota_BC6C11/RESULT = {"EnergyTotal":{"Total":[64.989,0.000,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001]}}
18:24:01.108 CMD: EnergyTotal3 0
18:24:01.117 MQT: stat/tasmota_BC6C11/RESULT = {"EnergyTotal":{"Total":[64.989,3.129,0.001],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001]}}
18:24:07.446 CMD: EnergyTotal 0
18:24:15.460 MQT: tele/tasmota_BC6C11/SENSOR = {"Time":"2022-11-10T18:24:15","ENERGY":{"TotalStartTime":"2022-11-10T18:24:08","Total":[64.989,3.129,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001],"Period":[0,0,0],"Power":[0,0,2],"ApparentPower":[9,0,26],"ReactivePower":[9,0,26],"Factor":[0.04,0.00,0.08],"Frequency":[50,50,50],"Voltage":[240,240,240],"Current":[0.04,0.00,0.11]}}
18:24:33.168 CMD: EnergyTotal 2 0
18:24:35.460 MQT: tele/tasmota_BC6C11/SENSOR = {"Time":"2022-11-10T18:24:35","ENERGY":{"TotalStartTime":"2022-11-10T18:24:35","Total":[64.989,3.129,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001],"Period":[0,0,0],"Power":[1,0,2],"ApparentPower":[9,0,26],"ReactivePower":[9,0,26],"Factor":[0.06,0.00,0.08],"Frequency":[50,50,50],"Voltage":[240,241,241],"Current":[0.04,0.00,0.11]}}
18:24:39.446 CMD: EnergyTotal 0 0
18:24:39.454 MQT: stat/tasmota_BC6C11/RESULT = {"EnergyTotal":{"Total":[0.000,3.129,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001]}}
18:24:41.599 CMD: EnergyTotal 0
18:24:41.608 MQT: stat/tasmota_BC6C11/RESULT = {"EnergyTotal":{"Total":[0.000,3.129,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001]}}

**Reply by FARKIr** [reply]
18:29:13.346 CMD: EnergyTotal
18:29:13.354 MQT: stat/tasmota_BC6C11/RESULT = {"EnergyTotal":{"Total":[64.989,3.129,2.020],"Yesterday":[0.000,0.000,0.000],"Today":[0.000,0.000,0.001]}}

**Reply by sfromis** [reply]
As your hardware is not Sonoff Pow R2, it **cannot be the same problem**. Don't hijack unrelated threads to ask questions.

**Reply by FARKIr** [reply]
I HAVE WHOLE CHIP ESP8266

**Comment by kellerst** [comment]
![power metering](https://github.com/arendst/Tasmota/assets/103080976/cf02e961-e503-4916-a963-bf729ae74761)

How can I set the value ?
Today 1.9 Wh > 3700 Wh
Total 191.569 kWh > 555.000 kWh
And can I send frequency 49.94 with MQTT ?

Build on Feb 9 2024 13:55:22 version 1.17.455
Thx

**Reply by sfromis** [reply]
See the commands: https://tasmota.github.io/docs/Commands/#power-monitoring

**Reply by kellerst** [reply]
Thank you very much, I know the link. I ask for an example Command so I can copy and paste. For example set value for energy today or energy total or energy yesterday ?
Edit: Ok, I saw in the log command CMD is not working:

LOG: Info:CMD:[WebApp Cmd 'EnergyYesterday1 3oooooo' Result] Unknown command
Info:CMD:[WebApp Cmd 'EnergyExport1' Result] Unknown command

![CommandLine1](https://github.com/arendst/Tasmota/assets/103080976/5548a5d0-5b8a-4874-94c7-2b354af1d33e)
![PowerMonitoringHA1](https://github.com/arendst/Tasmota/assets/103080976/0b8d2366-9cd8-4c92-9024-79c09ae565f8)
I´m sorry, its OpenBeken don´t Tasmota :(

**Comment by kellerst** [comment]
![SetEnergyCounterDone](https://github.com/arendst/Tasmota/assets/103080976/bc91074b-9999-40fd-b2a7-f03e2a72571f)
![Set energy total](https://github.com/arendst/Tasmota/assets/103080976/92aa2960-63da-4bf2-ab33-f8d88adcc1b2)
The Code for OpenBekenCommand:
 
EnergyCntReset 557000

Example for 557 kWh 
:)

**Comment by marsic-84** [comment]
How to reset "**tariff1**" and "**tariff2**" values???
